File: Portfolio/3.1P - Clock Class/Clock/Clock/Clock.cs
using System;

namespace Clock
{
    public class Clock
    {
        private Counter _seconds;
        private Counter _minutes;
        private Counter _hours;

        public Clock() 
        {
            _seconds = new Counter("Seconds");
            _minutes = new Counter("Minutes");
            _hours = new Counter("Hours");
        }

        private void TrackSecondsValue()
        {
            if (_seconds.Tick < 59)
            {
                _seconds.Increment();
            }
            else
            {
                _seconds.Reset();
                _minutes.Increment();
            }
        }

        private void TrackMinutesValue()
        {
            if (_minutes.Tick > 59)
            {
                _minutes.Reset();
                _hours.Increment();
            }
        }

        private void TrackHoursValue()
        {
            if (_hours.Tick > 23)
            {
                _hours.Reset();
            }
        }

        public void ClockTrack()
        {
            TrackSecondsValue();
            TrackMinutesValue();
            TrackHoursValue();
        }

        public string Time
        {
            get 
            {
                return $"{_hours.Tick:D2}:{_minutes.Tick:D2}:{_seconds.Tick:D2}"; 
            }
        }

        public void ClockReset()
        {
            _seconds.Reset();
            _minutes.Reset();
            _hours.Reset();
        }
    }
}

File: Portfolio/11.1P - Clock in Another Language📕/Clock/Clock.rb
$stdout.sync = true

module Clock
    class Program
        def main()
            clock = Clock.new()
            index = 0

            while (true)
                clock.clockTrack()
                puts(clock.time)

                sleep(1)

                system("clear") || system("cls")              # This line is to clear screen 
                index += 1

                if (index >= 3600)
                    break
                end
            end
        end
    end

    public
    class Counter
        attr_accessor :name, :tick
    
        public
        def initialize(name)
            @tick = 0
            @name = name
        end
    
        public
        def increment()
            @tick += 1
        end
    
        public
        def reset()
            @tick = 0
        end
    end
       
    public
    class Clock
        def initialize()
        @seconds = Counter.new("Seconds")
        @minutes = Counter.new("Minutes")
        @hours = Counter.new("Hours")
        end
    
        private
        def trackSecondsValue()
            if (@seconds.tick < 59)
                @seconds.increment()
            else
                @seconds.reset()
                @minutes.increment()
            end
        end
    
        private
        def trackMinutesValue()
            if (@minutes.tick > 59)
                @minutes.reset()
                @hours.increment()
            end
        end
    
        private
        def trackHoursValue()
            if (@hours.tick > 23)
                @hours.reset()
            end
        end
    
        public
        def clockTrack()
            trackSecondsValue()
            trackMinutesValue()
            trackHoursValue()
        end
    
        public
        def time
            format("%02d:%02d:%02d", @hours.tick, @minutes.tick, @seconds.tick)
        end
    
        public
        def clockReset()
            @seconds.reset()
            @minutes.reset()
            @hours.reset()
        end
    end
end

Clock::Program.new.main()